root/usr/lib64/ruby/vendor_ruby/discovery.rb

```diff
@@ -166,10 +166,10 @@ def upload(uri = discover_server, type = proxy_type, custom_facts = {})
     return
   end
   if uri.host.nil? or uri.port.nil?
-    log_err "Foreman URI host or port was not specified, cannot continue"
+    log_err "Server or proxy URI host or port was not specified, cannot continue"
```

Comment:
Then why change them at all?
I'm fine with leaving it either way and changing it only if we also change the url to something branded or a neutral name that works for both upstream and downstream.